submissions/m-ruslan/friends-app/src/sorter.js

```diff
@@ -0,0 +1,143 @@
+let users = [];
+let sorted = [];
+let filtered = [];
+let nameSearched = [];
+let ageSearched = [];
+const state = {
+  sortBy: "",
+  sortOrder: "",
+  filterGenderBy: "",
+  searchByName: "",
+  ageFrom: "",
+  ageTo: "",
+};
+
+let drawResult = null;
+
+const getValueOrDatasetValue = (parentElement, inputGroupName, datasetKey) => {
+  const checked = parentElement.querySelector(
+    `input[name=${inputGroupName}]:checked`
+  );
+  if (checked === null) return "";
+  return datasetKey === undefined ? checked.value : checked.dataset[datasetKey];
+};
```

Comment:
IMO, you should know when you will get `value` and when `data-*`. For now, this code looks like "I don't know what I want, just get me something" :)

You should avoid such stuff. because they *can* create unclear data flow.

Reply:
@zonzujiro thank you, I will take it into account in future